product_profile/models/product.py

```diff
@@ -0,0 +1,276 @@
+                    self.check_useless_key_in_vals(new_vals, key)):
+                new_vals.pop(key)
+        # super call must be after check_useless_key_in_vals() call
+        # because we compare value before and after write
+        res = super(ProductProfile, self).write(new_vals)
+        if new_vals:
+            for rec in self:
+                products = self.env['product.product'].search(
+                    [('profile_id', '=', rec.id)])
+                _logger.info(" >>> %s Products updating after updated '%s' pro"
+                             "duct profile" % (len(products), rec.name))
+                data = products._get_profile_data({'profile_id': rec.id})
+                products.write(data)
+
+        return res
+
+    @api.model
+    def check_useless_key_in_vals(self, vals, key):
+        """ If replacing values are the same than in db, we remove them.
+            Use cases:
+            1/ if in edition mode you switch a field
+               from value A to value B and then go back to value A
+               then save form, field is in vals whereas it shouldn't.
+            2/ if profile data are in csv file there are processed
+               each time module containing csv is loaded
+            we remove field from vals to minimize impact on products
+        """
+        comparison_value = self[key]
+        if self._fields[key].type == 'many2one':
+            comparison_value = self[key].id
+        elif self._fields[key].type == 'many2many':
+            comparison_value = [(6, False, self[key].ids), ]
+        return vals[key] == comparison_value
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form',
+                        toolbar=False, submenu=False):
+        """ Display a warning for end user if edit record """
+        res = super(ProductProfile, self).fields_view_get(
+            view_id=view_id, view_type=view_type, toolbar=toolbar,
+            submenu=submenu)
+        if view_type == 'form':
+            style = 'alert alert-warning oe_text_center oe_edit_only'
+            alert = etree.Element('h2', {'class': style})
+            alert.text = (_("If you update this profile, all products "
+                            "using this profile could also be updated. "
+                            "Changes can take a while."))
+            doc = etree.XML(res['arch'])
+            doc[0].addprevious(alert)
+            res['arch'] = etree.tostring(doc, pretty_print=True)
+        return res
+
+
+class ProductMixinProfile(models.AbstractModel):
+    _name = 'product.mixin.profile'
+
+    @api.model
+    def _get_profile_fields(self):
+        fields_to_exclude = set(get_profile_fields_to_exclude())
+        return [field for field in self.env['product.profile']._fields.keys()
+                if field not in fields_to_exclude]
+
+    @api.model
+    def _get_profile_data(self, values, filled_fields=None):
+        # Note for migration to v9
```

Comment:
Please rename the method _get_profile_data(self, values, filled_fields=None) => _get_vals_from_profile(self, values)